From 2b8f2b1926dbd0e7fada446c5f7a5258bbf1026a Mon Sep 17 00:00:00 2001
From: Justin Hammond <justin@dynam.ac>
Date: Wed, 7 Jun 2023 09:29:11 +0000
Subject: [PATCH] force useGLES on riscv

---
 toolkit/xre/glxtest.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolkit/xre/glxtest.cpp b/toolkit/xre/glxtest.cpp
index 509675cc44..bfd72c238d 100644
--- a/toolkit/xre/glxtest.cpp
+++ b/toolkit/xre/glxtest.cpp
@@ -537,7 +537,7 @@ static bool get_egl_gl_status(EGLDisplay dpy,
   PFNGLGETSTRING glGetString =
       cast<PFNGLGETSTRING>(eglGetProcAddress("glGetString"));
 
-#if defined(__arm__) || defined(__aarch64__)
+#if defined(__arm__) || defined(__aarch64__) || defined(__riscv)
   bool useGles = true;
 #else
   bool useGles = false;
