From fb783e737f3f9bc2884c9de62241533e52d9d06d Mon Sep 17 00:00:00 2001
From: Justin Hammond <justin@dynam.ac>
Date: Mon, 5 Jun 2023 22:05:51 +0800
Subject: [PATCH] force useGLes for riscv64

---
 toolkit/xre/glxtest.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/toolkit/xre/glxtest.cpp b/toolkit/xre/glxtest.cpp
index 509675cc44..5e1150e5ef 100644
--- a/toolkit/xre/glxtest.cpp
+++ b/toolkit/xre/glxtest.cpp
@@ -537,8 +537,9 @@ static bool get_egl_gl_status(EGLDisplay dpy,
   PFNGLGETSTRING glGetString =
       cast<PFNGLGETSTRING>(eglGetProcAddress("glGetString"));
 
-#if defined(__arm__) || defined(__aarch64__)
+#if defined(__arm__) || defined(__aarch64__) || defined(__riscv64__)
   bool useGles = true;
+#error haha
 #else
   bool useGles = false;
 #endif
-- 
2.40.1

