From eb4079e3993ee4d62ab1e9afeb14536bf6ea8980 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Wed, 12 Apr 2023 18:41:58 +0200
Subject: [PATCH] Include missing cmath header

---
 src/wayland/outputmanagement_v2_interface.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/wayland/outputmanagement_v2_interface.cpp b/src/wayland/outputmanagement_v2_interface.cpp
index bed5dcd648..2a6069d9d6 100644
--- a/src/wayland/outputmanagement_v2_interface.cpp
+++ b/src/wayland/outputmanagement_v2_interface.cpp
@@ -17,6 +17,7 @@
 
 #include "qwayland-server-kde-output-management-v2.h"
 
+#include <cmath>
 #include <optional>
 
 using namespace KWin;
-- 
2.37.2

